rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -2375,6 +2375,35 @@ public UserData getCurrentUserData() {
         return userData;
     }
 
+    @Override
+    public void releaseDanglingBusyNodes(List<NodeSet> busyNodesWithTask) {
+        Set<String> busyNodesUrlWithTask = new HashSet<>();
+
+        for (NodeSet nodeSet : busyNodesWithTask) {
+            busyNodesUrlWithTask.addAll(nodeSet.getAllNodesUrls());
+        }
+
+        List<RMNode> busyNodesToRelease = findDanglingBusyNodes(busyNodesUrlWithTask);
+
+        nodesCleaner.cleanAndRelease(busyNodesToRelease);
+    }
+
+    private List<RMNode> findDanglingBusyNodes(Set<String> busyNodesUrlWithTask) {
```

Comment:
Why not call this method findBusyNodes ?

Reply:
findBusyNodes is not explicit enough in my opinion. findBusyNodesWithoutTask?

Reply:
Renamed into `findBusyNodesNotIn(Set<String> nodesUrlToNotRelease)`